roles/docker_swarm_node/tasks/existing_join_tokens.yml

```diff
@@ -1,21 +1,65 @@
-- name: fetch swarm status of all nodes
-  docker_swarm:
-    state: inspect
-  register: swarm_info
+- name: Fetch swarm status for managers
+  command: docker info --format '{{ "{{json .Swarm}}" }}'
+  register: docker_swarm_status
+  when: "'docker_swarm_managers' in group_names"
   become: true
 
-- name: grab first join token and manager to join
+# - debug: var=docker_swarm_status.stdout
+
+- name: Extract Swarm.LocalNodeState
+  shell: echo '{{ docker_swarm_status.stdout }}' | tr -d '\000-\037' | jq -r '.LocalNodeState // empty'
+  register: local_node_state
+  changed_when: false
+
+- name: Set swarm_local_node_state fact
+  set_fact:
+    swarm_local_node_state: "{{ local_node_state.stdout }}"
+
+- name: Extract Swarm.ControlAvailable
+  shell: echo '{{ docker_swarm_status.stdout }}' | tr -d '\000-\037' | jq -r '.ControlAvailable // empty'
```

Comment:
You could avoid all of the dependencies on tr and jq and any differences in how they may work between releases or operating systems by using go templates here instead, see: https://docs.docker.com/engine/cli/formatting/